src/components/CurrentRefinedValues/CurrentRefinedValues.js

```diff
@@ -5,8 +5,13 @@ import Template from '../Template.js';
 import {isSpecialClick} from '../../lib/utils.js';
 import map from 'lodash/collection/map';
 import cloneDeep from 'lodash/lang/cloneDeep';
+import {isEqual} from 'lodash';
 
 class CurrentRefinedValues extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return !isEqual(this.props.refinements, nextProps.refinements);
```

Comment:
Shouldn't we ensure we only manipulate immutable props so we could replace the deep `isEqual` by a regular `===`?